main/com.tunein/manifest.tt

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 TuneIn, Inc.
+//           2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Redistribution and use in source and binary forms, with or
+// without modification, are permitted provided that the following
+// conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above
+//    copyright notice, this list of conditions and the following
+//    disclaimer in the documentation and/or other materials
+//    provided with the distribution.
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+// INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+// OF THE POSSIBILITY OF SUCH DAMAGE.
+
+class @com.tunein
+#_[thingpedia_name="Tunein"]
+#_[thingpedia_description="Listen to internet radio"]
+#_[canonical="radio"]
+#[license="BSD-3-Clause"]
+#[license_gplcompatible=true]
+#[subcategory="media"]
+{
+  import loader from @org.thingpedia.v2();
+  import config from @org.thingpedia.config.none();
+
+  monitorable query list search_channel(in req query: String
```

Comment:
The design of this manifest is not quite right. You want to structure this like a database table: a single query called "channel" (or maybe "station") with only input parameters. Search is performed as filtering in ThingTalk
(this is the same design as Yelp, Spotify, etc.).

So "search by name" is:
```
@com.tunein.channel(), id =~ "koit";
``

"search by frequency" is:
```
@com.tunein.channel(), frequency == 96.5MHz;
```
(or just `96.5`)

"list by genre" is:
```
@com.tunein.channel(), contains~(genre, "pop");
```
etc.

"list local channels" would be a special case of an optional input parameter, because of the default handling of current location and because you want to match locations by regions not by equality:
```
@com.tunein.channel(location=new Location("..."));
```

("list trending channels" could also be a special case of an optional input parameter, or a boolean output parameter, not clear)